src/main/java/ecdar/abstractions/Engine.java

```diff
@@ -52,11 +52,11 @@ public void setDefault(boolean aDefault) {
     }
 
     public String getEngineLocation() {
-        return backendLocation;
+        return engineLocation;
     }
 
-    public void setEngineLocation(String backendLocation) {
-        this.backendLocation = backendLocation;
+    public void setEngineLocation(String engineLocation) {
+        this.engineLocation = engineLocation;
```

Comment:
I am not entirely sure but is ``engineLocation`` a path to a executable?
Maybe we should do some validation on this argument eg. checking if the location points to an executable file.
Also, I think that the engine location maybe should be stored as a File and then ``getEngineLocation`` could return the absolute path to the file.

Reply:
The `engineLocation` can refer to both an executable and a remote IP address. The validation you suggest is done when the engine is constructed through the GUI (as this allows for error messages to be displayed at an appropriate time and node)
We basically assume that the location is valid at the time we get here, as it is stored as a string within the local preferences. Maybe adding some validation here could be a good idea if we start adding engines through other workflows, but the current setup makes it extraneous.